### PR TITLE
feat: add filter for active jobs in the dashboard

### PR DESCRIPTION
**Pull request by hmacr**
Closes #<issue>

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/triggerdotdev/trigger.dev/blob/main/CONTRIBUTING.md)
- [x] The PR title follows the convention.
- [x] I ran and tested the code works

---

## Testing

- Verified that the toggle filters the active jobs.
- Verified that all jobs are listed when the toggle is inactive 
---

## Changelog

- Add `useToggleFilter` which filters items based on `toggle` status.
- Use `useToggleFilter` in the `useFilterJobs` hook.

---

## Screenshots

https://github.com/triggerdotdev/trigger.dev/assets/132386067/e73df1e5-b2da-43cb-8cd6-1063bc305283

💯

### CODE REVIEW

**Comment by hmacr** [comment]
@matt-aitken @samejr PR for the feature discussed on Discord ([link](https://discord.com/channels/1066956501299777596/1066956844553207828/1161307734231236658)). Please review, thanks!

File: apps/webapp/app/components/primitives/Switch.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const variations = {
     container: "flex items-center gap-x-1.5 rounded hover:bg-slate-850 pr-1 py-[0.1rem] pl-1.5",
     root: "h-3 w-6",
     thumb: "h-2.5 w-2.5 data-[state=checked]:translate-x-2.5 data-[state=unchecked]:translate-x-0",
-    text: "text-xs text-slate-400 group-hover:text-slate-200 mt-0.5",
```
Comment (samejr) [inline apps/webapp/app/components/primitives/Switch.tsx:18]: @hmacr could you provide a screen grab of the toggle switch on the Integration page to check this margin change hasn't altered the layout of that Component?

File: apps/webapp/app/routes/_app.orgs.$organizationSlug.projects.$projectParam._index/route.tsx [hunk 2]
```diff
@@ -106,6 +112,13 @@ export default function Page() {
                           onChange={(e) => setFilterText(e.target.value)}
                           autoFocus
                         />
+                        <Switch
```
Comment (samejr) [inline apps/webapp/app/routes/_app.orgs.$organizationSlug.projects.$projectParam._index/route.tsx:114]: @hmacr could you add `gap-x-2` to the parent of this to give some space between the items in this row?

CleanShot 2023-10-11 at 10 25 50

**Comment by samejr** [comment]
Hey @hmacr it's looking great. Only 2 small layout tweaks/checks from me 👍

**Comment by hmacr** [comment]
Thanks @samejr ! Have you left your review somewhere? I don't see it in this PR.

**Comment by samejr** [comment]
> Thanks @samejr ! Have you left your review somewhere? I don't see it in this PR.

I've left code comments inline. I can see them above in this thread - or you should be able to see them in the commits.

**Comment by hmacr** [comment]
@samejr Sorry, I don't find them in this thread / PR. Could you please check if the review has been submitted?


https://github.com/triggerdotdev/trigger.dev/assets/132386067/e68f110a-3042-4a18-9ecb-6d87a58d6f3e

**Review by samejr — CHANGES_REQUESTED** [review]

**Comment by samejr** [comment]
Sorry @hmacr totally forgot to submit the review. My bad.

File: apps/webapp/app/routes/_app.orgs.$organizationSlug.projects.$projectParam._index/route.tsx (hunk 2 above)
Comment (hmacr) [reply apps/webapp/app/routes/_app.orgs.$organizationSlug.projects.$projectParam._index/route.tsx:114]: @samejr Added the gap.

Screenshot 2023-10-11 at 3 36 04 PM

File: apps/webapp/app/components/primitives/Switch.tsx (hunk 1 above)
Comment (hmacr) [reply apps/webapp/app/components/primitives/Switch.tsx:18]: @samejr Here's the screen grab showing that both the toggle switches are center aligned vertically.

https://github.com/triggerdotdev/trigger.dev/assets/132386067/89e95194-3ae1-4af4-b1ba-72478db958bd

**Review by samejr — APPROVED** [review]

**Pull request merged**